sds_data_manager/stacks/api_gateway_stack.py

```diff
@@ -0,0 +1,84 @@
+from aws_cdk import Environment, Stack
+from aws_cdk import aws_apigateway as apigw
+from aws_cdk import aws_certificatemanager as acm
+from aws_cdk import aws_route53 as route53
+from aws_cdk import aws_route53_targets as targets
+from constructs import Construct
+
+
+class ApiGateway(Stack):
+    """Sets up api gateway, creates subdomains, and creates methods that
+    are linked to the lambda function.
+
+    An example of the format of the url: https://dev.imap-mission.com/query
+    """
+
+    def __init__(
+        self,
+        scope: Construct,
+        construct_id: str,
+        sds_id: str,
+        lambda_functions: dict,
+        env: Environment,
+        hosted_zone: route53.IHostedZone = None,
+        certificate: acm.ICertificate = None,
+        use_custom_domain: bool = False,
+        environment_name: str = "dev",
+        **kwargs,
+    ) -> None:
+        super().__init__(scope, construct_id, env=env, **kwargs)
+
+        # Define subdomains
+        subdomains = lambda_functions.keys()
+
+        # Create a single API Gateway
+        api = apigw.RestApi(
+            self,
+            f"api-RestApi-{sds_id}",
+            rest_api_name=f"api-RestApi-{sds_id}",
+            description="This service serves as my API Gateway.",
+            deploy_options=apigw.StageOptions(stage_name=f"{sds_id}"),
+            endpoint_types=[apigw.EndpointType.REGIONAL],
+        )
+
+        # Define a custom domain
+        if use_custom_domain:
+            custom_domain = apigw.DomainName(
+                self,
+                f"api-DomainName-{sds_id}",
+                domain_name=f"{environment_name}.imap-mission.com",
```

Comment:
```suggestion
                domain_name=f"api.{sds_id}.imap-mission.com",
```

What about using the sds_id as our subdomain identifier? "environment" can mean a lot of different things, so it isn't clear what you're asking the user for to me.

Another note is that I think we'll want a few different things here for routing off of this location, so I think we should also probably put `api.*` (or some other qualifier) in the domain route so we know this is where our APIGateway is living. (We will also have the `dev.imap-mission.com` HTML site being served, possibly some `opensearch.dev.imap-mission.com` database routing too.